tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,77 @@
+import json
+import subprocess
+import urllib.request
+from time import sleep
+
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(15))
+def test_dcgm_exporter():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "http://localhost:9400/metrics"
+    dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+    subprocess.run(f"sudo snap start {dcgm_exporter_service}".split(), check=True)
+
+    result = subprocess.run(
+        f"sudo snap services {dcgm_exporter_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    assert " active" in result.stdout.strip(), "dcgm-exporter service is not active"
```

Comment:
I think we can decouple the activation of the exporter with the check of the http point (maybe declare this activation on a separate function on conftest), so the retry is clear if it's failing because of the exporter is not activating, or if it's because the http endpoint is failing. Moreover, tests should be independent and there is a dependency of this test running before `test_dcgm_port_configs` because otherwise the exporter won't be up and running

Reply:
Sure, but there will still be a dependency, `test-dcgm-port-configs` should be run after all other tests as it changes the ports of the exporter and nv-host engine, so other tests would fail.

Reply:
Should I revert the ports back in the `test-dcgm-port-configs`?

Reply:
> Should I revert the ports back in the test-dcgm-port-configs?

Yes, tests should not be dependent on order and this has the potential to break tests in the future